src/solving/algorithm/bfs.rs

```diff
@@ -46,37 +47,38 @@ impl BFSSolver {
 
     fn bfs_iteration(
         &mut self,
-        current_board: OwnedBoard,
-        current_path: Vec<BoardMove>,
+        current_board: &OwnedBoard,
+        current_path: &Vec<BoardMove>,
```

Comment:
Writing `&Vec` instead of `&[_]` involves a new object where a slice will do. See [clippy::ptr_arg](https://rust-lang.github.io/rust-clippy/master/index.html#/ptr_arg)
Instead of cloning a `Vec`, you can pass a slice reference and construct a new `Vec` from it